lib/ansible/plugins/inventory/gcp_compute.py

```diff
@@ -339,23 +343,22 @@ def parse(self, inventory, loader, path, cache=True):
             self._sanitize_group_name = self._legacy_script_compatible_group_sanitization
 
         # get user specifications
-        if 'zones' in config_data:
-            if not isinstance(config_data['zones'], list):
-                raise AnsibleParserError("Zones must be a list in GCP inventory YAML files")
+        zones = self.get_option('zones')
+        config_data['zones'] = zones
+        if not isinstance(zones, list):
```

Comment:
also not needed, add `type: list` to configuration definition, nothing should be added back to config_data

Reply:
> also not needed, add type: list to configuration definition

I can do that

> nothing should be added back to config_data

I can't do that. The entire solution here pivots on putting `"service_account_file"` in config data, because I don't know how it is used for the request auth. I would be guessing at what you want this to look like. I need more detail.

Reply:
nothing should use config_data, its only meant for 'extras not covered by get_option', but that is a preexisting problem here